circuit/environment/src/helpers/counter.rs

```diff
@@ -16,17 +16,17 @@
 use crate::*;
 use snarkvm_fields::PrimeField;
 
-use std::{mem, rc::Rc};
+use std::{mem, sync::Arc};
```

Comment:
Why the change to Arc? I recall Howard mention several times that this part of the code was explicitly designed to be only used in single-threaded runtimes, so Arc might give the wrong impression.

Reply:
The switch is needed in order for those objects to work during the `Assignment` stage or afterwards. `Arc` is perfectly viable as a substitute for `Rc` even if no parallelism is involved, and we can always make an appropriate comment indicating why it is not an `Rc`.